Thanks for the patch!

Sorry, there's a bit of bitrot going on in that harness at the moment. Here's the latest failure after applying the change in this PR:

https://github.com/cilium/cilium/actions/runs/20529083512/job/59433731578?pr=43504#step:4:618

```
pkg/policy/l4_filter_test_fuzz.go:487:15: undefined: OriginForTest
```

It looks like https://github.com/cilium/cilium/commit/0399e2f47d951 didn't take into account that this function was being used by fuzzers outside the package(?). Perhaps we can work around that or just revert that commit.

For broader context also, the most recent attempt to fix up CIFuzz was over here: https://github.com/cilium/cilium/pull/41780 . I haven't had a chance to really sit down and try to dig through and figure out how to get these tests back up. Previously I've had a lot more success with `make fuzz` than with these external harnesses (though I just noticed that was broken, so I've fixed it up here: https://github.com/cilium/cilium/pull/43555).

--- a/test/fuzzing/oss-fuzz-build.sh
+++ b/test/fuzzing/oss-fuzz-build.sh
@@ -10,15 +10,21 @@ set -eu
 
 
 ln -s "$SRC"/cilium/pkg/policy/distillery_test{,_fuzz}.go
+ln -s "$SRC"/cilium/pkg/policy/l4_filter_deny_test{,_fuzz}.go
 ln -s "$SRC"/cilium/pkg/policy/l4_filter_test{,_fuzz}.go
 ln -s "$SRC"/cilium/pkg/policy/l4_test{,_fuzz}.go
 ln -s "$SRC"/cilium/pkg/policy/mapstate_test{,_fuzz}.go
+ln -s "$SRC"/cilium/pkg/policy/origin_test{,_fuzz}.go
+ln -s "$SRC"/cilium/pkg/policy/repository_deny_test{,_fuzz}.go
 ln -s "$SRC"/cilium/pkg/policy/repository_test{,_fuzz}.go
 ln -s "$SRC"/cilium/pkg/policy/resolve_test{,_fuzz}.go
 ln -s "$SRC"/cilium/pkg/policy/resolve_deny_test{,_fuzz}.go
 ln -s "$SRC"/cilium/pkg/policy/rule_test{,_fuzz}.go
 ln -s "$SRC"/cilium/pkg/policy/selectorcache_test{,_fuzz}.go
 
+# Allow Go to download dependencies not present in the vendor directory.
+# This is needed for go-118-fuzz-build/testing which is required by compile_native_go_fuzzer.
+export GOFLAGS="-mod=mod"
 
 compile_native_go_fuzzer github.com/cilium/cilium/pkg/container/bitlpm FuzzUint8 FuzzUint8
 compile_native_go_fuzzer github.com/cilium/cilium/pkg/fqdn/matchpattern FuzzMatchpatternValidate FuzzMatchpatternValidate

